### PR TITLE
header: skip empty blocks

### LINKED ISSUE #470: service/header: bridge node should short-circuit storing of empty blocks
Currently, we process empty blocks like any other blocks, but we should account for this special by excluding them from being written, again and again. 

Empty blocks are always equal, and there can be many of them through the life of the chain, but there is no point in writing the same data many times and wasting io.

### PR DESCRIPTION
**Pull request by vgonkivs**
Fixes #470

### CODE REVIEW

File: service/header/core_listener.go [hunk 1]
```diff
@@ -79,6 +79,11 @@ func (cl *CoreListener) listen(ctx context.Context, sub <-chan *types.Block) {
 
 			eh, err := MakeExtendedHeader(ctx, b, comm, vals, cl.dag)
 			if err != nil {
+				if err == errEmptyBlock {
+					log.Debugw("core-listener: ", err)
```
Comment (renaynay) [inline service/header/core_listener.go:83]: We should probably log more details about the block here (height, time received, etc.)

**Review by adlerjohn — COMMENTED** [review]
Is this actually needed? Without looking into the implementation, shouldn't `ComputeShares()`, `ipld.PutData()`, and `NewDataAvailabilityHeader()` have some sort of fast-path for empty blocks already?

**Comment by adlerjohn** [comment]
Need a fix to the pending changelog.

**Review by renaynay — COMMENTED** [review]
👍🏻 

Can you also add a test for this?

**Comment by vgonkivs** [comment]
> Is this actually needed? Without looking into the implementation, shouldn't ComputeShares(), ipld.PutData(), and NewDataAvailabilityHeader() have some sort of fast-path for empty blocks already?

`ipld.PutData()` returns an error for empty block. But do we need an extra call for  `ComputeShares()` and   `ipld.PutData()` if we can rely on txs length?

@adlerjohn

**Comment by vgonkivs** [comment]
Added test.

File: CHANGELOG-PENDING.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 ### FEATURES
 
 ### IMPROVEMENTS
-
+- [header: bridge node should short-circuit storing of empty blocks #480](https://github.com/celestiaorg/celestia-node/pull/480) [@vgonkivs](https://github.com/vgonkivs)
```
Comment (renaynay) [inline CHANGELOG-PENDING.md:10]:
```suggestion
- [header: bridge node avoids duplicate storing of empty blocks #480](https://github.com/celestiaorg/celestia-node/pull/480) [@vgonkivs](https://github.com/vgonkivs)
```

**Review by renaynay — APPROVED** [review]
❤️

**Review by adlerjohn — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]
Sweet! `b.Txs` is a good solution

**Pull request merged**